packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1591,9 +1584,16 @@ export class MergeTree {
         refSeq = this.collabWindow.currentSeq,
         clientId = this.collabWindow.clientId) {
         const seg = refPos.getSegment();
-        if (seg && seg.parent) {
-            const offset = !seg.removedSeq ? refPos.getOffset() : 0;
-            return offset + this.getPosition(seg, refSeq, clientId);
+        if (seg?.parent === undefined) {
+            return DetachedReferencePosition;
+        }
+        if (refPos.isLeaf()) {
+            return this.getPosition(refPos, refSeq, clientId);
+        }
+        if (refTypeIncludesFlag(refPos, ReferenceType.Transient)
```

Comment:
in regard to the above, i've also changed this function to only return position if the local reference is linked. this allows persevering the segment and offset, and allows someone to manually get the original position if they need it.